### ISSUE TITLE
Donation marked as "Not eligible for Giftaid" included in batch

### ISSUE THREAD

**Issue by tapashdatta** [mattwire/uk.co.compucorp.civicrm.giftaid#7]
As title says,
Although a specific donation marked as "No" in eligible for Giftaid , it gets included in batch when created a new batch.

**Comment by mattwire** [comment]
@tapashdatta Can you confirm if this is still an issue with 3.3.3 please?

**Comment by tapashdatta** [comment]
@mattwire Yes, It's still the case. It happens when a person has a open declaration with yes, and 4 years and a donation from backend (ie cash or check) that is marked NO. It could be the case that a person may want a particular donation  to be NOT giftaidable, but still want the declaration to be kept open.

**Comment by mattwire** [comment]
Please check with 3.3.7

**Comment by tapashdatta** [comment]
@mattwire Yes, It seems to be working now. Also taking financial types rules into consideration. Thanks

**Comment by tapashdatta** [comment]
@mattwire  Seems I spoke too soon.  Still  There are donation taken from backend that are getting rejected although the financial type is enabled.

Also some of the disable financial types donation, if they are processed from backend, they are getting included in the batch.

**Comment by tapashdatta** [comment]
@mattwire This still is happening for marked as "No" for individual donation, but has open declaration.
EDIT: ... and Just because the financial types were enabled in global settings.
EDIT1: I think if eligibilty criteria placed at the individual donation level, this issue would be resolved.

**Comment by mattwire** [comment]
@tapashdatta Can you test https://github.com/mattwire/uk.co.compucorp.civicrm.giftaid/pull/20
